src/commands/openapi/upload.ts

```diff
@@ -147,10 +155,20 @@ export default class OpenAPIUploadCommand extends BaseCommand<typeof OpenAPIUplo
       this.handleAPIRes(res),
     );
 
-    // if the current slug already exists, we'll use PUT to update it. otherwise, we'll use POST to create it
-    const method = existingAPIDefinitions?.data?.some((d: { filename: string }) => d.filename === filename)
-      ? 'PUT'
-      : 'POST';
+    const matchingAPIDefinition = existingAPIDefinitions?.data?.find((d: { filename: string; legacy_id: string }) => {
+      if (this.flags['legacy-id']) {
+        return d?.legacy_id === this.flags['legacy-id'];
```

Comment:
Do we want to throw an error if a user passes in `legacy-id` and we don't find a matching API definition?

Reply:
yeah we should probably throw an error

Reply:
i'm not sure if the error should be thrown here (it might make sense to do it after this `.find` statement?) but agreed that we should throw an error if they pass the legacy flag and it doesn't match anything!